pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,9 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		originalContainerImageNames := getContainerImageNames(spec.Template)
+		defer setContainerImageNames(spec.Template, originalContainerImageNames)
```

Comment:
Ah - I though you had it as well.  OK, we'll wait for @bparees / @smarterclayton , though I just realized I forgot another comment from @kargakis up above which I think is worth doing.  Will address after scrum.